encoders/firebase-encoders-reflective/src/main/java/com/google/firebase/encoders/reflective/ReflectiveInitializer.java

```diff
@@ -35,11 +36,11 @@
               + " does not define a no-argument constructor. If you are using ProGuard, make "
               + "sure these constructors are not stripped.");
     } catch (IllegalAccessException e) {
-      throw new RuntimeException(e);
+      throw new EncodingException(e.toString());
```

Comment:
We have to preserve the cause chain to make the error more useful, I would suggest something like this in all cases below:
```java
} catch (IllegalAccessException | InstantiationException | InvocationTargetException e) {
  throw new EncodingException("Could not instantiate type " + classToken, e);
}
```

not the second parameter to the exception constructor.

Reply:
pls also add the `e` argument to the catch block above.

Reply:
Multi-catch requires API level 19.

Reply:
No really, if you set language level to java 8, the desugarer will kick in and desugar to java 6.

i.e.  see https://github.com/firebase/firebase-android-sdk/blob/master/firebase-common/firebase-common.gradle#L49